go/hyperkit.go

```diff
@@ -355,6 +358,9 @@ func (h *HyperKit) buildArgs(cmdline string) {
 	if h.DiskImage != "" {
 		a = append(a, "-s", fmt.Sprintf("2:0,virtio-blk,%s", h.DiskImage))
 	}
+	if h.VSock {
+		a = append(a, "-s", fmt.Sprintf("3,virtio-sock,guest_cid=3,path=%s", h.StateDir))
```

Comment:
You may want to consider appending a subdir to `h.StateDir` so the various Unix sockets are segregated from the rest of the state stuff and avoid potential conflicts.

Otherwise this all LGTM